rioxarray/rioxarray.py

```diff
@@ -230,7 +216,79 @@ def crs(self):
         """:obj:`rasterio.crs.CRS`:
             The projection of the dataset.
         """
-        return self._crs
+        raise NotImplementedError
+
+    def set_crs(self, input_crs):
```

Comment:
So you decided not to use the `@property` version of this?

Reply:
I think having it as a method could potentially be more flexible, allow method chaining, and it would match the interface in `geopandas`. See: https://github.com/geopandas/geopandas/pull/747#issuecomment-398913889

Thoughts?

Reply:
Ok, but it seems like most of their argument is related to method chaining and your `set_crs` doesn't return anything. That said, I like the idea of consistency.

Reply:
> set_crs doesn't return anything

Oh, hah :man_facepalming:. Good catch. I should fix that. Thanks!

Reply:
Chaining should work now.

Reply:
How does this work with xarray accessors? You're returning the xarray object, but does that object have this same `_crs` attribute that was just set? To match geopandas does this need to set the CRS on a copy of the xarray object and then return that copy?

Reply:
> You're returning the xarray object, but does that object have this same _crs attribute that was just set?

It seems to be the case in the tests.

```python
>>> test_da = xarray.DataArray(
...     numpy.zeros((5, 5)),
...     dims=("y", "x"),
...     coords={"y": numpy.arange(1, 6), "x": numpy.arange(2, 7)},
... )
>>> test_da.rio.crs
>>> ds2 = test_da.rio.set_crs(4326)
>>> ds2.rio.crs
CRS.from_epsg(4326)
>>> test_da.rio.crs
CRS.from_epsg(4326)

```

> To match geopandas does this need to set the CRS on a copy of the xarray object and then return that copy?

Hmmm, I guess an `inplace=` argument would be useful. It seems the default they are thinking of is `inplace=True` as making a copy when setting the attribute is not entirely necessary.

Reply:
Just made the change.